content/docs/testing-environments.md

```diff
@@ -0,0 +1,57 @@
+---
+id: testing-environments
+title: Testing environments
+permalink: docs/testing-environments.html
+---
+
+<!-- This document is intended for folks who are comfortable with Javascript, and have probably written tests with it. It acts as a reference for the differences in testing environments for React components, and how those differences affect the tests that they write. This document also assumes a slant towards web-based react-dom components, but has notes for other renderers. -->
+
+Testing environments can affect the way you write and run tests for React Components. 
+
+This document goes through the major factors that can affect your environment, the implications of those decisions, and recommendations for some scenarios.
+
+### Test frameworks {#test-frameworks}
+
+Test frameworks like [Jest](https://jestjs.io/), [mocha](https://mochajs.org/), [ava](https://github.com/avajs/ava) let you write test suites as regular Javascript, and run them as part of your development process. Additionally, test suites are run as part of workflows like builds, CI/CD, etc.
+
```

Comment:
It's probably worth calling out here that some setups like Create React App already include a test runner. https://facebook.github.io/create-react-app/docs/running-tests

Maybe even before this section. We should avoid the situation where people spend an hour configuring Jest in a CRA project (which happened before).